tests/codegen/ccode/scripts/functions.py

```diff
@@ -0,0 +1,43 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+# pylint: disable=unused-variable
+
+def helloworld():
+    print('hello world')
+
+#$ header function incr(int)
+def incr(x):
+    x = x + 1
+
+#$ header function decr(int) results(int)
+def decr(x):
+    y = x - 1
+    return y
+
+#$ header function f1(int, int, int) results(int)
+def f1(x, n=2, m=3):
+    y = x - n*m
+    return y
+
+#$ header function f2(int, int) results(int)
+def f2(x, m=None):
+    if m is None:
+        y = x + 1
+    else:
+        y = x - 1
+    return y
+
+y = decr(2)
+z = f1(1)
+
+z1 = f2(1)
+z2 = f2(1, m=0)
+
+helloworld()
+
+# TODO add messages. for the moment there's a bug in Print
```

Comment:
Can you explain this TODO? Or is it copied and pasted from fcode? If it's copy pasted then can you look into it and see if there is still a problem

Reply:
yes it's copied and pasted from the code, I tested it and the printing seems to be working normally in `Fortran` 
but o get a compilation error in `C`
for example this python code : 
```
#$ header function f2(int, int) results(int)
def f2(x, m=None):
    if m is None:
        y = x + 1
    else:
        y = x - 1
    return y


z1 = f2(1)
z2 = f2(1, m=0)

# TODO add messages. for the moment there's a bug in Print
print(z1)
print(z2)
```
generate this main C code which fail in the compilation : 

```
int main()
{
    int64_t z1;
    int64_t z2;


    z1 = f2(1, NULL);
    z2 = f2(1, &m);
    /*TODO add messages. for the moment there's a bug in Print*/
    printf("%ld\n", z1);
    printf("%ld\n", z2);
    return 0;
}
```
```
error: use of
      undeclared identifier 'm'
```

Reply:
Ok. This error is not related to the print so could you remove the TODO message in both files please. In C it looks like there is either a problem with passing values to optional parameters. I.e:
```
#$ header function f2(int, int) results(int)
def f2(x, m=None):
    ...


z2 = f2(1, 0)
```
Or there is a problem with keyword arguments. I.e:
```
#$ header function f2(int, int) results(int)
def f2(x, m):
    ...


z1 = f2(1, m=0)
```
Or there is a problem when combining the 2. I.e.:
```
#$ header function f2(int, int) results(int)
def f2(x, m = None):
    ...


z1 = f2(1, m=0)
```

Could you work out which and open an issue for this please

Reply:
i tested those cases, and it look like there is a problem with keyword arguments.

Reply:
ok. Thanks :)

Reply:
@rakati